test/parallel/test-dgram-udp6-link-local-address.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+const common = require('../common');
+if (!common.hasIPv6)
+  common.skip('no IPv6 support');
+
+const assert = require('assert');
+const dgram = require('dgram');
+
+const os = require('os');
+const ifaces = os.networkInterfaces();
+
+// Try to an interface with an IPv6 link local address
+const address = Object.keys(ifaces)
+
+// Convert the interface names to the interface arrays of addresses
+      .map(
+        (ifname) => ifaces[ifname]
+          .map((ifobject) => Object.assign({ ifname: ifname }, ifobject)))
+
+  // Flatten the array of arrays
+      .reduce((prev, curr) => prev.concat(curr))
+
+  // Filter out all non-IPv6 addresses
+      .filter((iface) => iface.family === 'IPv6')
+
+  // Map the interface into its address, with interface ID
+      .map((iface) => iface.address + '%' + iface.ifname)
+
+  // Return the first one with a link-local address
+      .find((addr) => addr.startsWith('fe80:'));
```

Comment:
This could be simplified, if I read the logic right?
```js
function linklocal() {
  for (const [ifname, entries] of Object.entries(os.networkInterfaces()))
    for (const {address, family} of entries)
      if (family === 'IPv6' && address.startsWith('fe80:'))
        return address + '%' + ifname;
}
const address = linklocal();
```
Although I'm not sure picking a random interface is right for a test.  I have several interfaces with link-local addresses configured on my system but I'm fairly sure only lo + lo0 accept traffic.

Reply:
Hmm.  I guess we should also test that the interface is up.  IMHO relying on `lo` or `lo0` is a bad practice, since there is no guarantee that future systems will have such an interface name.

Reply:
Well, not `UP` or not, but at least on my Mac only `status: active` interfaces seem to accept traffic.  But then it is only so that those have a link local IPv6 address.  The `inactive` ones don't have any IPv4 or IPv6 addresses.  Hmm.  The tap/tun -like interface `utun0` seems to be an exception here, it does have an IPv6 address but it does not have any `status`, at least not in the `ifconfig` output.

I guess my knowledge of the BSD / Linux networking internals is lacking here.  Any suggestions?